pkg/installer/dnsmasq/dnsmasq_config.go

```diff
@@ -0,0 +1,47 @@
+package dnsmasq
+
+import (
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"sigs.k8s.io/yaml"
+)
+
+type DNSConfig struct {
+	APIIntIP                 string
+	IngressIP                string
+	GatewayDomains           []string
+	GatewayPrivateEndpointIP string
+}
+
+func CreatednsmasqIgnitionFiles(bootstrapAsset *bootstrap.Bootstrap, installConfig *installconfig.InstallConfig, dnsConfig DNSConfig) error {
+	dnsmasqIgnConfig, err := Ignition3Config(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.IngressIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, true)
+	if err != nil {
+		return err
+	}
+	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, dnsmasqIgnConfig.Storage.Files...)
```

Comment:
This will be fine once the fork is replaced with the upstream installer, but in the meantime this will result in two files with the same filename ending up in the list (one from the fork, and one from here). Since the unit tests look at the first one, it means this new code is not being verified by the tests until we switch to the upstream installer.

By changing this to search for the file(s) we are looking for and replacing them if they exist (and appending only if they don't) then we can incrementally replace the fork's implementation and test as we go.

This applies to all the places we are appending stuff to the ignition.

Reply:
I have too many commits, I forgot the replaceOrAppend function. Will add it.